webserver/cors/src/main/java/io/helidon/webserver/cors/CORSSupport.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.webserver.cors;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.helidon.config.Config;
+import io.helidon.webserver.Handler;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.ServerRequest;
+import io.helidon.webserver.ServerResponse;
+import io.helidon.webserver.Service;
+
+/**
+ * A Helidon service and handler implementation that implements CORS, for both the application and for built-in Helidon
+ * services (such as OpenAPI and metrics).
+ * <p>
+ *     The caller can set up the {@code CORSSupport} in a combination of these ways:
+ * </p>
+ *     <ul>
+ *         <li>from a {@link Config} node supplied programmatically,</li>
+ *         <li>from one or more {@link CrossOriginConfig} objects supplied programmatically, each associated with a path to which
+ *         it applies, and</li>
+ *         <li>by setting individual CORS-related attributes on the {@link Builder} (which affects the CORS behavior for the
+ *         {@value Aggregator#PATHLESS_KEY} path).</li>
+ *     </ul>
+ * <p>
+ *     See the {@link Builder#build} method for how the builder resolves conflicts among these sources.
+ * </p>
+ * <p>
+ *     If none of these sources is used, the {@code CORSSupport} applies defaults as described for
+ *     {@link CrossOriginConfig}.
+ * </p>
+ *
+ */
+public abstract class CORSSupport implements Service, Handler {
```

Comment:
As in MP, please rename to camel case: `CorsSupport` 
Please do so for `CorsSupportHelper`, `CorsSupportSe`, `SeReqeustAdapter`, `SeResponseAdapter`